Android/app/src/main/java/io/github/project_travel_mate/utilities/CustomAnalogClock.java

```diff
@@ -0,0 +1,279 @@
+package io.github.project_travel_mate.utilities;
+
+
+import android.content.Context;
+import android.content.res.Resources;
+import android.content.res.TypedArray;
+import android.graphics.Canvas;
+import android.graphics.drawable.Drawable;
+import android.os.Handler;
+import android.support.annotation.DrawableRes;
+import android.util.AttributeSet;
+import android.view.View;
+import android.view.animation.Animation;
+import android.view.animation.AnimationSet;
+import android.view.animation.LinearInterpolator;
+import android.view.animation.RotateAnimation;
+
+import io.github.project_travel_mate.R;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.TimeZone;
+
+
+/**
+ * A widget that displays the time as a 12-at-the-top 24 hour analog clock. By
+ * default, it will show the current time in the current timezone. The displayed
+ * time can be set using {@link #setTime(long)} and and
+ * {@link #setTimezone(TimeZone)}.
+ *
+ * @author <a href="mailto:steve@staticfree.info">Steve Pomeroy</a>
+ */
+public class CustomAnalogClock extends View {
+
+    public static boolean is24;
+    public static boolean hourOnTop;
+    private final ArrayList<DialOverlay> mDialOverlay = new ArrayList<DialOverlay>();
+    private Calendar mCalendar;
+    private Drawable mFace;
+    private int mDialWidth;
+    private float mSizeScale = 1f;
+    private int mRadius;
+    private int mDialHeight;
+    private int mBottom;
+    private int mTop;
+    private int mLeft;
+    private int mRight;
+    private boolean mSizeChanged;
+    private HandsOverlay mHandsOverlay;
+    private boolean mAutoUpdate;
+
+    public CustomAnalogClock(Context context, AttributeSet attrs, int defStyle) {
+        super(context, attrs, defStyle);
+        handleAttrs(context, attrs);
+    }
+
+    public CustomAnalogClock(Context context, AttributeSet attrs) {
+        super(context, attrs);
+        handleAttrs(context, attrs);
+    }
+
+    public CustomAnalogClock(Context context) {
+        super(context);
+        init(context);
+    }
+
+    public CustomAnalogClock(Context context, boolean defaultWatchFace) {
```

Comment:
add javadoc style comments to all the newly added functions

Reply:
Added

Reply:
not added for all yet

Reply:
??